packages/web/src/components/user-list/UserListV2.tsx

```diff
@@ -0,0 +1,210 @@
+import { useCallback } from 'react'
+
+import { useCurrentUserId } from '@audius/common/api'
+import { ID, User, FollowSource } from '@audius/common/models'
+import { profilePageActions, usersSocialActions } from '@audius/common/store'
+import { route } from '@audius/common/utils'
+import { Flex, FollowButton } from '@audius/harmony'
+import { range } from 'lodash'
+import InfiniteScroll from 'react-infinite-scroller'
+import { useDispatch } from 'react-redux'
+
+import ArtistChip from 'components/artist/ArtistChip'
+import { MountPlacement } from 'components/types'
+import * as unfollowConfirmationActions from 'components/unfollow-confirmation-modal/store/actions'
+import { useIsMobile } from 'hooks/useIsMobile'
+import { push } from 'utils/navigation'
+
+import { UserListItemSkeleton } from './UserListItemSkeleton'
+
+const { profilePage } = route
+const { setNotificationSubscription } = profilePageActions
+
+const SCROLL_THRESHOLD = 400
+
+type SkeletonItem = {
+  _loading: true
+  user_id: string
+  tag?: string
+}
+
+type UserListV2Props = {
+  /**
+   * The list of users to display
+   */
+  data: User[] | undefined
+  /**
+   * Whether there are more users to load
+   */
+  hasNextPage: boolean
+  /**
+   * Whether we're loading more users
+   */
+  isFetchingNextPage: boolean
+  /**
+   * Whether we're loading the initial data
+   */
+  isLoading: boolean
+  /**
+   * Function to load more users
+   */
+  fetchNextPage: () => void
+  /**
+   * Callback when a user navigates away
+   */
+  onNavigateAway?: () => void
+  /**
+   * Callback after following a user
+   */
+  afterFollow?: () => void
+  /**
+   * Callback after unfollowing a user
+   */
+  afterUnfollow?: () => void
+  /**
+   * Optional callback before clicking artist name
+   */
+  beforeClickArtistName?: () => void
+  /**
+   * Optional function to get scroll parent element
+   */
+  getScrollParent?: () => HTMLElement | null
+  /**
+   * Optional user ID to show support for (used in top supporters list)
+   */
+  showSupportFor?: ID
+  /**
+   * Optional user ID to show support from (used in supporting list)
+   */
+  showSupportFrom?: ID
+}
+
+export const UserListV2 = ({
+  data,
+  hasNextPage,
+  isFetchingNextPage,
+  isLoading,
+  fetchNextPage,
+  onNavigateAway,
+  afterFollow,
+  afterUnfollow,
+  beforeClickArtistName,
+  getScrollParent,
+  showSupportFor,
+  showSupportFrom
+}: UserListV2Props) => {
+  const dispatch = useDispatch()
+  const isMobile = useIsMobile()
+  const { data: currentUserId } = useCurrentUserId()
+
+  const handleLoadMore = useCallback(() => {
+    if (hasNextPage && !isFetchingNextPage) {
+      fetchNextPage()
```

Comment:
iirc, the fetchNextPage from useInfiniteQuery internally checks hasNextPage (and maybe isFetchingNextPage)